### DISCUSSION TITLE
Asynchronize `Swarm<T>` initialization and fix NineChronicles Agent Swarm<T> initialization

### DISCUSSION THREAD

**Discussion by tkiapril** [planetarium/libplanet#2303] [General]
As of libplanet 0.41.1, initialization of `Swarm<T>` is not fully asynchronous, and uses `Task.GetAwaiter().GetResult()` which blocks the main thread and causes context deadlock when used in certain circumstances. As a workaround, in NineChronicles in `Init()` method in `Agent.cs`, `Swarm<T>` initialization is done inside a `Task.Run()` invocation to isolate the blocking method from the main thread after libplanet 0.41.1 bump. After a API rework, this should be fixed.

The workaround is as follows: from this:
```csharp
            _swarm = new Swarm<NCAction>(
                blocks,
                privateKey,
                appProtocolVersion: appProtocolVersion,
                host: host,
                listenPort: port,
                iceServers: iceServers,
                differentAppProtocolVersionEncountered: DifferentAppProtocolVersionEncountered,
                trustedAppProtocolVersionSigners: trustedAppProtocolVersionSigners);
```
To this:
```csharp
            // FIXME: this should be changed to reflect libplanet API change after it is reworked.
            // For context, refer to https://github.com/planetarium/libplanet/discussions/2303.
            var initSwarmTask = Task.Run(() => new Swarm<NCAction>(
                blocks,
                privateKey,
                appProtocolVersion: appProtocolVersion,
                host: host,
                listenPort: port,
                iceServers: iceServers,
                differentAppProtocolVersionEncountered: DifferentAppProtocolVersionEncountered,
                trustedAppProtocolVersionSigners: trustedAppProtocolVersionSigners));

            initSwarmTask.Wait();
            _swarm = initSwarmTask.Result;
```

**Comment by longfin** [comment]
>  As a workaround, in NineChronicles in Init() method in Agent.cs, Swarm<T> initialization is done inside a Task.Run() invocation to isolate the blocking method from the main thread after libplanet 0.41.1 bump

@tkiapril Could you introduce that workaround here? 🙏

**Reply by tkiapril** [reply]
Updated the OP!

**Comment by longfin** [comment]
[As commented previously](https://github.com/planetarium/libplanet/pull/2215#issuecomment-1218147153), I still think that we have two choices. 

- Introduce new asynchronous factory method other than synchronous constructor
- Admit "unintilalzed state" (😇 again)
